libavutil/hwcontext_qsv.c

```diff
@@ -1063,6 +1063,34 @@ static int qsv_transfer_data_from(AVHWFramesContext *ctx, AVFrame *dst,
     if (ret < 0)
         return ret;
 
+    /* According to MSDK spec for mfxframeinfo, "Width must be a multiple of 16.
+     * Height must be a multiple of 16 for progressive frame sequence and a
+     * multiple of 32 otherwise.", so allign all frames to 16 before downloading. */
+    if (ctx->height & 15 || dst->linesize[0] & 15) {
+        AVFrame *tmp_frame;
+        tmp_frame = av_frame_alloc();
+        if (!tmp_frame)
+            return AVERROR(ENOMEM);
+        ret = av_frame_ref(tmp_frame, dst);
+        if (ret < 0)
+            return ret;
```

Comment:
call `av_frame_free(&tmp_frame)` before returning an error.

Reply:
Done. Thanks

Reply:
I am sorry calling `av_frame_free(&tmp_frame)` here is not needed here.